desc/profiles.py

```diff
@@ -811,12 +811,14 @@ class SplineProfile(_Profile):
         - `'catmull-rom'`: C1 cubic centripetal "tension" splines
     name : str
         name of the profile
+    df : array-like
+        Optional. Values of the function derivative at knot locations.
 
     """
 
     _io_attrs_ = _Profile._io_attrs_ + ["_knots", "_method"]
 
-    def __init__(self, values=None, knots=None, method="cubic2", name=""):
+    def __init__(self, values=None, knots=None, method="cubic2", name="", df=None):
```

Comment:
> not crazy about adding this option, since it basically overrides method, and if we wanted to do something like this df would need to be part of params so it can be optimized as well. - Rory

Reply:
Another possible solution would be to make a new profile class specifically for hermite splines where the derivative is required as part of params but method is always cubic. Reasoning as follows:

- For the existing spline class, all the different `method` options are just different ways of estimating the derivative. If the derivatives are supplied then this is redundant.
- Adding `fx` as an optional parameter means that sometimes the number of parameters is `n`, other times `2*n` which I feel like will cause problems if it gets used in optimization.